Tests/PrefixDurationTests.swift

```diff
@@ -0,0 +1,151 @@
+
+        var results = [Int]()
+
+        cancellable = subject
+            .prefix(duration: 0.5)
+            .sink(receiveCompletion: { _ in expectation.fulfill() },
+                  receiveValue: { results.append($0) })
+
+        DispatchQueue.main.asyncAfter(deadline: .now() + 1.5) {
+            subject.send(1)
+        }
+
+        wait(for: [expectation], timeout: 2)
+
+        XCTAssertTrue(results.isEmpty)
+    }
+
+    func testFinishedInWindow() {
+        let subject = PassthroughSubject<Int, Never>()
+        let expectation = XCTestExpectation()
+
+        var results = [Subscribers.Completion<Never>]()
+
+        cancellable = subject
+            .prefix(duration: 0.5)
+            .sink(receiveCompletion: { results.append($0); expectation.fulfill() },
+                  receiveValue: { _ in })
+
+        DispatchQueue.main.asyncAfter(deadline: .now() + 0.25) {
+            subject.send(completion: .finished)
+        }
+
+        wait(for: [expectation], timeout: 2)
+
+        XCTAssertEqual(results, [.finished])
+    }
+
+    private enum AnError: Error {
+        case someError
+    }
+
+    func testErrorInWindow() {
+        let subject = PassthroughSubject<Int, AnError>()
+        let expectation = XCTestExpectation()
+
+        var results = [Subscribers.Completion<AnError>]()
+
+        cancellable = subject
+            .prefix(duration: 0.5)
+            .sink(receiveCompletion: { results.append($0); expectation.fulfill() },
+                  receiveValue: { _ in })
+
+        DispatchQueue.main.asyncAfter(deadline: .now() + 0.25) {
+            subject.send(completion: .failure(.someError))
+        }
+
+        wait(for: [expectation], timeout: 2)
+
+        XCTAssertEqual(results, [.failure(.someError)])
+    }
+
+    func testErrorEventOutsideWindowDoesntAffectFinishEvent() {
+        let subject = PassthroughSubject<Int, AnError>()
+        let expectation = XCTestExpectation()
+
+        var results = [Subscribers.Completion<AnError>]()
+
+        cancellable = subject
+            .prefix(duration: 0.5)
+            .sink(receiveCompletion: { results.append($0); expectation.fulfill() },
+                  receiveValue: { _ in })
+
+        DispatchQueue.main.asyncAfter(deadline: .now() + 0.75) {
+            subject.send(completion: .failure(.someError))
+        }
+
+        wait(for: [expectation], timeout: 2)
+
+        XCTAssertEqual(results, [.finished])
+    }
+
+    func testMultipleEventsInAndOutOfWindow() {
+        let subject = PassthroughSubject<Int, Never>()
+        let expectation = XCTestExpectation()
+
+        var results = [Int]()
+        var completions = [Subscribers.Completion<Never>]()
+
+        cancellable = subject
+            .prefix(duration: 0.5)
+            .sink(receiveCompletion: { completions.append($0); expectation.fulfill() },
+                  receiveValue: { results.append($0) })
+
+        subject.send(1)
+        subject.send(2)
+        subject.send(3)
```

Comment:
Can we have one of these delayed within the window? Also maybe move the test up after the single event test, so it's a bit chronological